### ISSUE TITLE
Update ValidatingWebhookConfiguration

### ISSUE THREAD

**Issue by olix0r** [linkerd/linkerd2#5797]
Installing Linkerd onto recent Kubernetes versions, we see a warning like:

```
Warning: admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
```

We should update the webhook resources as needed.

**Comment by alpeb** [comment]
That was addressed in #5603, but wasn't back-ported into stable.

**Comment by olix0r** [comment]
@alpeb ah gotcha
